### PR TITLE
homework - 5

### PR DESCRIPTION
**Pull request by rana123bilal**
Done:
- [1] - Added View Movie details feature upon clicking on any of the movie card in the list.
-  [2] - Use useState and useEffect hooks;
- [3] - Use useContext hook to avoid props drilling.

Not done:
-[1] - nothing

Final mark: 5

### CODE REVIEW

File: src/components/cards/media-card.js [hunk 1]
```diff
@@ -48,6 +48,10 @@ export default function MediaCard({ id, title, image, genre, year }) {
   const handleMouseOut = () => {
     setIsHovering(false);
```
Comment (Vladlan) [inline src/components/cards/media-card.js:49]: Why you named this fn `handleMouseOut`? It is totally clear that it is `setIsHovering`

File: src/components/movie/view-movie.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+function ViewMovie() {
+  const { setViewMovieDetails, movieId } = useContext(DataContext);
+  const [viewMovie, setViewMovie] = useState(null);
+
+  useEffect(() => {
+    const foundMovie = DUMMY_DATA.find((movie) => movie.id === movieId);
+    setViewMovie(foundMovie);
+  }, [movieId, viewMovie]);
+
+  function onclickSearch() {
```
Comment (Vladlan) [inline src/components/movie/view-movie.js:16]: onclickSearch -> closeViewMovieDetails

File: src/components/cards/media-card.js (hunk 1 above)
Comment (Vladlan) [reply src/components/cards/media-card.js:49]: Please correct all function names in a way:
1. Name should describe what this function is doing (not when like onClickHandler)
2. So in the end you will be able to read your JSX like `<CardMedia onClick={toggleCard}/> `

File: src/components/cards/media-card.js [hunk 3]
```diff
@@ -48,6 +48,10 @@ export default function MediaCard({ id, title, image, genre, year }) {
   const handleMouseOut = () => {
```
Comment (Vladlan) [inline src/components/cards/media-card.js:48]: setIsHovering(true) ????? Set is hovering what? maybe toggleKebabMenu(true)?

**Review by Vladlan — CHANGES_REQUESTED** [review]
Also please add custom hook of any kind (it is totally up to you).

File: src/components/cards/media-card.js (hunk 3 above)
Comment (Vladlan) [reply src/components/cards/media-card.js:48]: Please remove setIsHovering at all and use css for this (`.class_name:hover .class {...`)

**Comment by Vladlan** [comment]
Also 
1. please check again https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/%5BCDP%5D-Home-Task-%E2%80%93-React-v1?node-id=0%3A1 and correct UI in view-movie.js
2. Add font according to figma
3. Check paddings in all forms (inputs)
4. Fix bug with jumping card on hover

To summarise - make UI look good (since you are a frontend engineer it should be by default)

File: src/components/cards/media-card.js (hunk 1 above)
Comment (rana123bilal) [reply src/components/cards/media-card.js]: 1 :- Removed handMouseout and handmouseOver.
2 :- Changed function names accordingly.

File: src/components/cards/media-card.js [hunk 4]
```diff
@@ -48,6 +48,10 @@ export default function MediaCard({ id, title, image, genre, year }) {
...
     setIsHovering(false);
   };
+  const cardHandler = () => {
```
[the remark this answers is not in the document]
Comment (rana123bilal) [reply src/components/cards/media-card.js]: I changed name of this function.

File: src/components/cards/media-card.js [hunk 5]
```diff
@@ -7,16 +7,16 @@ import MenuItem from "@mui/material/MenuItem";
...
 
 export default function MediaCard({ id, title, image, genre, year }) {
-
   const {
     setMovieIdForDeleteEdit,
     openEditMovieModal,
     openDeleteMovieModal,
     setOpenEditMovieModal,
     setOpenDeleteMovieModal,
+    setViewMovieDetails,
```
[the remark this answers is not in the document]
Comment (rana123bilal) [reply src/components/cards/media-card.js:15]: Removed this verb.

File: src/components/movie/view-movie.js (hunk 2 above)
Comment (rana123bilal) [reply src/components/movie/view-movie.js]: Changed.

File: src/components/cards/media-card.js (hunk 3 above)
Comment (rana123bilal) [reply src/components/cards/media-card.js]: Removed isHovering states and use Pseudo selectors instead.

**Review by Vladlan — APPROVED** [review]

**Pull request merged**